docs/repo/dependencies-node-roadmap.md

```diff
@@ -4,19 +4,21 @@ This document describes, at a high level, the design and implementation of the D
 
 Let's start at the top and work our way down. There are two fundamentally different paths that dependency information takes from the Project System to Solution Explorer. Direct dependencies go through CPS, whereas transitive dependencies generally go through graph node providers.
 
-## The CPS View of Dependencies
+## Overview
+
+### The CPS View of Dependencies
 
 On the CPS side the project tree is composed of instances of the `IProjectTree` and `IProjectItemTree` interfaces. `IProjectTree` captures the structure of the tree (e.g. it has properties to access the parent and child nodes) and the "UI" aspects of the node--name, icons, visibility, etc. An `IProjectItemTree` captures all of that but also represents a concrete item within the project like a file, assembly reference, or NuGet package.
 
 An `IProjectTree` is immutable. When a part of the tree needs to be updated, we need to replace it and form a new tree.
 
 Components wishing to add items to the project tree we must implement and export the `IProjectTreeProvider` interface. In practice we implement the interface by deriving from the abstract `ProjectTreeProviderBase` class provided by CPS. In general, we receive events/messages from CPS detailing changes to the project, generated an updated `IProjectTree`, and then pass it back to CPS via `ProjectTreeProviderBase.SubmitTreeUpdateAsync`.
```

Comment:
I know it's not part of your PR but there are two typos here:

- project tree ~~we~~ must implement
- changes to the project, generate~~d~~ an updated

I can open a separate PR for these if you prefer.